test/dynamics/signals/test_pulse_to_signals.py

```diff
@@ -205,15 +218,59 @@ def test_multiple_channels_with_gaps(self):
         samples2 = np.append(np.zeros(15), np.append(gauss_samples, np.zeros(10)))
         samples3 = np.append(np.zeros(20), np.append(gauss_samples, np.zeros(5)))
 
-        self.assertAllClose(signals[0].samples, samples0, atol=1e-14, rtol=1e-14)
-        self.assertAllClose(signals[1].samples, samples1, atol=1e-14, rtol=1e-14)
-        self.assertAllClose(signals[2].samples, samples2, atol=1e-14, rtol=1e-14)
-        self.assertAllClose(signals[3].samples, samples3, atol=1e-14, rtol=1e-14)
+        self.assertAllClose(signals[0].samples, samples0, atol=1e-7, rtol=1e-7)
+        self.assertAllClose(signals[1].samples, samples1, atol=1e-7, rtol=1e-7)
+        self.assertAllClose(signals[2].samples, samples2, atol=1e-7, rtol=1e-7)
+        self.assertAllClose(signals[3].samples, samples3, atol=1e-7, rtol=1e-7)
         self.assertTrue(signals[0].carrier_freq == 5.0)
         self.assertTrue(signals[1].carrier_freq == 3.1)
         self.assertTrue(signals[2].carrier_freq == 0.0)
         self.assertTrue(signals[3].carrier_freq == 4.0)
 
+    def test_get_samples(self):
+        """Test get samples of Pulse not get_waveform but get_samples function."""
+        gauss_get_waveform_samples = (
+            pulse.Gaussian(duration=5, amp=0.983, sigma=2.0).get_waveform().samples
+        )
+        gauss_get_samples = get_samples(Gaussian(duration=5, amp=0.983, sigma=2.0))
+        self.assertTrue(isinstance(gauss_get_samples, np.ndarray))
+        self.assertAllClose(gauss_get_samples, gauss_get_waveform_samples, atol=1e-7, rtol=1e-7)
+
+    def test_lru_cache_expr(self):
+        """Test lru_cache of lru_cache_expr function."""
+        gauss_envelop = Gaussian(duration=5, amp=0.983, sigma=2.0).envelope
+        self.assertTrue(
+            _lru_cache_expr(gauss_envelop, Array.default_backend())
+            is _lru_cache_expr(gauss_envelop, Array.default_backend())
+        )
+
+
+class TestJaxGetSamples(QiskitDynamicsTestCase, TestJaxBase):
+    """Tests get_samples function by using Jax."""
+
+    def setUp(self):
+        """Set up gaussian waveform samples for comparison."""
+        self.gauss_get_waveform_samples = (
+            pulse.Gaussian(duration=5, amp=0.983, sigma=2.0).get_waveform().samples
+        )
+
+    def test_get_samples_with_jax(self):
+        """Test get samples of Pulse not get_waveform but get_samples function in Jax case."""
+        gauss_get_samples = get_samples(Gaussian(duration=5, amp=0.983, sigma=2.0))
+        self.assertTrue(isinstance(gauss_get_samples, jnp.ndarray))
+        self.assertAllClose(
+            gauss_get_samples, self.gauss_get_waveform_samples, atol=1e-7, rtol=1e-7
+        )
+
+    def test_jit_get_samples(self):
+        """Test compiling to get samples of Pulse."""
+
+        def jit_func(amp):
+            return get_samples(Constant(100, amp))
+
+        jit_samples = jax.jit(jit_func)(0.1)
```

Comment:
We expected this function will pass without using static_argnums after merging https://github.com/Qiskit/qiskit-terra/pull/9002